### PR TITLE
feat:[#3127] Add access type to the research products

### LINKED ISSUE #3127: Add "Access type" to the Pin to project functionality
Acceptance criteria:

- [ ] Add the "Access type" field to the Pinning details according to Figma

https://www.figma.com/file/wFzD4qH6SwpfCybte3pLlb/EOSC-DRAFTS?type=design&node-id=2050-57653&mode=design&t=LS2pZZx4fcQVQsa2-4

### PR DESCRIPTION
**Pull request by goreck888**
Add best_access_rights to the research products table Displaying access type in details

Changes:
- access type badge look
- styles: access type look in projects panel
- styles: fix for collapse icon

Closes #3127

### CODE REVIEW

File: db/migrate/20231110091337_add_access_type_to_research_products.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# frozen_string_literal: true
+
+class AddAccessTypeToResearchProducts < ActiveRecord::Migration[6.1]
+  def change
+    add_column :research_products, :best_access_right, :string
```
Comment (wujuu) [inline db/migrate/20231110091337_add_access_type_to_research_products.rb:5]: shouldn't the column name be "access_type"?

Comment (wujuu) [reply db/migrate/20231110091337_add_access_type_to_research_products.rb:5]: also in other places - why it's named "best_access_right"?

Comment (goreck888) [reply db/migrate/20231110091337_add_access_type_to_research_products.rb:5]: Thanks to it I just consume a JSON object from the SearchService request without additional part of code for mapping JSON and DB Object fields.

Comment (wujuu) [reply db/migrate/20231110091337_add_access_type_to_research_products.rb:5]: makes sense

**Comment by Marcelinna** [comment]
@goreck888 
image
Add resource to this project should redirected to all catalogs

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**